doc/langref.html.in

```diff
@@ -4915,87 +4925,92 @@ test "if boolean" {
+test "if error union" {
+    // If expressions test for errors.
+    // Note the |err| capture on the else.
 
-    const a: ?u32 = 0;
+    const a: anyerror!u32 = 0;
     if (a) |value| {
         try expect(value == 0);
-    } else {
+    } else |err| {
+        _ = err;
         unreachable;
     }
 
-    const b: ?u32 = null;
-    if (b) |_| {
+    const b: anyerror!u32 = error.BadValue;
+    if (b) |value| {
+        _ = value;
         unreachable;
-    } else {
-        try expect(true);
+    } else |err| {
+        try expect(err == error.BadValue);
     }
 
-    // The else is not required.
+    // The else and |err| capture is strictly required.
     if (a) |value| {
         try expect(value == 0);
-    }
+    } else |_| {}
 
-    // To test against null only, use the binary equality operator.
-    if (b == null) {
-        try expect(true);
+    // To check only the error value, use an empty block expression.
+    if (b) |_| {} else |err| {
+        try expect(err == error.BadValue);
     }
 
     // Access the value by reference using a pointer capture.
-    var c: ?u32 = 3;
+    var c: anyerror!u32 = 3;
     if (c) |*value| {
-        value.* = 2;
+        value.* = 9;
+    } else |_| {
+        unreachable;
     }
 
     if (c) |value| {
-        try expect(value == 2);
-    } else {
+        try expect(value == 9);
+    } else |_| {
         unreachable;
     }
 }
+      {#code_end#}
+      {#header_open|if with Optionals#}
 
-test "if error union" {
-    // If expressions test for errors.
-    // Note the |err| capture on the else.
+      {#code_begin|test|test_if_optionals#}
+const expect = @import("std").testing.expect;
 
-    const a: anyerror!u32 = 0;
+test "if optional" {
+    // If expressions test for null.
+
+    const a: ?u32 = 0;
     if (a) |value| {
         try expect(value == 0);
-    } else |err| {
-        _ = err;
+    } else {
         unreachable;
     }
 
-    const b: anyerror!u32 = error.BadValue;
-    if (b) |value| {
-        _ = value;
+    const b: ?u32 = null;
+    if (b) |_| {
         unreachable;
-    } else |err| {
-        try expect(err == error.BadValue);
+    } else {
+        try expect(true);
     }
 
-    // The else and |err| capture is strictly required.
+    // The else is not required.
     if (a) |value| {
         try expect(value == 0);
-    } else |_| {}
+    }
 
-    // To check only the error value, use an empty block expression.
-    if (b) |_| {} else |err| {
-        try expect(err == error.BadValue);
+    // To test against null only, use the binary equality operator.
+    if (b == null) {
+        try expect(true);
     }
 
     // Access the value by reference using a pointer capture.
-    var c: anyerror!u32 = 3;
+    var c: ?u32 = 3;
     if (c) |*value| {
-        value.* = 9;
-    } else |_| {
-        unreachable;
+        value.* = 2;
     }
 
     if (c) |value| {
-        try expect(value == 9);
-    } else |_| {
+        try expect(value == 2);
+    } else {
```

Comment:
The diff is terrible but basically I took the two sections relating to optional and put them under their own heading.

Reply:
Oof, yeah the diff is not helping.   The new if-err-capture examples all look reasonable to me.

Reply:
I didn't add anything - it is just exactly as it was but split into two sections.  I built the docs and they work as expected locally.